components/maze_solver/include/maze_solver/direction.h

```diff
@@ -213,6 +213,27 @@ constexpr auto to_radians(Direction d) noexcept
     }
 }
 
+template <ExtendedFloatingPoint F = float>
+constexpr auto to_direction(const F &a) noexcept
+{
+    static constexpr F quarter_pi{std::numbers::pi_v<float> / 4};
```

Comment:
Incorrect type for pi:

```suggestion
    static constexpr F quarter_pi{std::numbers::pi_v<make_floating_point<F>> / 4};
```

But it would probably be more useful to have:

```cpp
    static constexpr auto pi = std::numbers::pi_v<make_floating_point<F>>;
```

or


```cpp
    static constexpr F pi{std::numbers::pi_v<make_floating_point<F>>};
```